File: laondry-order-service/internal/entity/order_item.go
package entity

import (
	"time"

	"github.com/google/uuid"
	"gorm.io/gorm"
)

type OrderItem struct {
	ID          uuid.UUID `gorm:"type:uuid;primary_key" json:"id"`
	OrderID     uuid.UUID `gorm:"type:uuid;not null;index" json:"order_id"`
	ServiceID   uuid.UUID `gorm:"type:uuid;not null;index" json:"service_id"`
	ServiceCode string    `gorm:"type:varchar(50);not null;index" json:"service_code"`
	ServiceName string    `gorm:"type:varchar(150);not null" json:"service_name"`
	WeightKg    *float64  `gorm:"type:decimal(8,2)" json:"weight_kg"`
	Qty         *int      `gorm:"type:int" json:"qty"`
	UnitPrice   float64   `gorm:"type:decimal(12,2);not null" json:"unit_price"`
	LineTotal   float64   `gorm:"type:decimal(12,2);not null" json:"subtotal"` // Mobile expects subtotal
    CreatedAt   time.Time `gorm:"not null" json:"created_at"`
    UpdatedAt   time.Time `gorm:"not null" json:"updated_at"`

	Order   *Order            `gorm:"foreignKey:OrderID;references:ID" json:"order,omitempty"`
	Service *Service          `gorm:"foreignKey:ServiceID;references:ID" json:"service,omitempty"`
	Addons  []OrderItemAddon  `gorm:"foreignKey:OrderItemID;constraint:OnDelete:CASCADE" json:"addons,omitempty"`
}

func (OrderItem) TableName() string {
	return "order_items"
}

func (oi *OrderItem) BeforeCreate(tx *gorm.DB) error {
	if oi.ID == uuid.Nil {
		oi.ID = uuid.New()
	}
	return nil
}

File: laondry-system-management-core-api/app/Models/OrderService/OrderItem.php
<?php

namespace App\Models\OrderService;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Concerns\HasUuids;

class OrderItem extends Model
{
    use HasFactory, HasUuids;

    protected $fillable = [
        'order_id',
        'service_id',
        'service_code',
        'service_name',
        'weight_kg',
        'qty',
        'unit_price',
        'line_total'
    ];
    protected $casts = [
        'weight_kg' => 'decimal:2',
        'unit_price' => 'decimal:2',
        'line_total' => 'decimal:2',
    ];

    public function order()
    {
        return $this->belongsTo(Order::class);
    }
    public function service()
    {
        return $this->belongsTo(Service::class);
    }
    public function addons()
    {
        return $this->hasMany(OrderItemAddon::class, 'order_item_id');
    }
}
